lib/src/realm_class.dart

```diff
@@ -376,32 +415,62 @@ class Realm implements Finalizable {
   /// Disclaimer: This method is mostly needed on Dart standalone and if not called the Dart program will hang and not exit.
   /// This is a workaround of a Dart VM bug and will be removed in a future version of the SDK.
   static void shutdown() => scheduler.stop();
+
+  void _throwIfFrozen() {
+    if (isFrozen) {
+      throw RealmError('Starting a write transaction on a frozen Realm is not allowed.');
+    }
+  }
 }
 
-/// @nodoc
+/// Provides a scope to safely write data to a [Realm]. Must be created using [Realm.beginWrite].
 class Transaction {
   Realm? _realm;
 
+  /// Returns whether the transaction is still active.
+  bool get isOpen => _realm != null;
+
   Transaction._(Realm realm) {
     _realm = realm;
-    realmCore.beginWrite(realm);
   }
 
+  /// Commits the changes to the Realm.
   void commit() {
-    if (_realm == null) {
-      throw RealmException('Transaction was already closed. Cannot commit');
-    }
+    _ensureOpen('commit');
 
     realmCore.commitWrite(_realm!);
-    _realm = null;
+    _closeTransaction();
+  }
+
+  /// Commits the changes to the Realm asynchronously.
+  /// Canceling the commit using the [cancellationToken] will not abort the transaction, but
+  /// rather resolve the future immediately with a [CancelledException].
+  Future<void> commitAsync({CancellationToken? cancellationToken}) async {
+    _ensureOpen('commit');
+
+    await realmCore.commitWriteAsync(_realm!, cancellationToken);
+
+    _closeTransaction();
```

Comment:
Should we ensure this is called even if the above throws?

Reply:
No, we discussed it in the PR comments - https://github.com/realm/realm-dart/pull/930#issuecomment-1276324855 and the next couple of comments. Ultimately, we don't want to close the transaction if committing fails because that would be the wrong behavior. The user should still be able to retry the commit, even if it's unlikely that it'll succeed.

Reply:
ok, I noticed the discussion but this comment was still open with a thumbs up. Hence asking.